### PR TITLE
Update rubocop

### PR DESCRIPTION
**Pull request by aaron-collier**
Note: This is a work in progress as I work through a few of the more complex exclusions here (which may stay as exclusions, there were initially 500+ violations).

This updates rubocop to 0.52.0 to address the reported vulnerability concern.

### CODE REVIEW

File: spec/runners/active_encode_derivatives_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 describe Hydra::Derivatives::ActiveEncodeDerivatives do
-  context '.create' do
+  context 'when create' do
```
Comment (mjgiarlo) [inline spec/runners/active_encode_derivatives_spec.rb]: I think this should change to `describe '.create' do` instead.

File: spec/services/remote_source_file_spec.rb [hunk 2]
```diff
@@ -5,14 +5,14 @@ class TestObject < ActiveFedora::Base
 end
 
 describe Hydra::Derivatives::RemoteSourceFile do
-  describe '.call' do
+  describe 'when called' do
```
Comment (mjgiarlo) [inline spec/services/remote_source_file_spec.rb]: I'm not getting this change. Does rubocop yell about `describe '.METHOD' do` now? I wouldn't expect a `'when...'` string with a `describe` (tend to see that with `context`s).

File: spec/services/tempfile_service_spec.rb [hunk 3]
```diff
@@ -21,8 +23,7 @@ def has_content?
 
   let(:file) { class_with_metadata_extraction.new(initialization_options) }
 
-  subject { described_class.new(file) }
-  context '#tempfile' do
+  context 'with a tempfile' do
```
Comment (mjgiarlo) [inline spec/services/tempfile_service_spec.rb]: Not `describe '#tempfile' do`?

Comment (aaron-collier) [reply spec/services/tempfile_service_spec.rb]: Yeah, rubocop is complaining that a `context` should always start with when/with/etc... I'm happy to exclude if we feel like it.

File: spec/services/remote_source_file_spec.rb (hunk 2 above)
Comment (aaron-collier) [reply spec/services/remote_source_file_spec.rb]: You're right, overzealous updates based on the context issue noted below. Changing back now.

File: spec/runners/active_encode_derivatives_spec.rb (hunk 1 above)
Comment (aaron-collier) [reply spec/runners/active_encode_derivatives_spec.rb]: done

File: spec/services/tempfile_service_spec.rb (hunk 3 above)
Comment (aaron-collier) [reply spec/services/tempfile_service_spec.rb]: @mjgiarlo you're right, updated.

**Pull request merged**